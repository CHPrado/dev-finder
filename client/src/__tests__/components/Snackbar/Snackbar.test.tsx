import React from "react";

import { mount, shallow } from "enzyme";
import toJson from "enzyme-to-json";

import { Snackbar } from "../../../components";

const message = "Test message";

describe("rendering component", () => {
  it("renders Snackbar component without crashing", () => {
    shallow(<Snackbar />);
  });
});

describe("passing props", () => {
  it("with open equals true, message and type props", () => {
    const wrapper = mount(
      <Snackbar open={true} message={message} type="success" />
    );

    expect(wrapper.props().open).toEqual(true);
    expect(wrapper.props().type).toEqual("success");
    expect(wrapper.props().message).toEqual(message);
    expect(wrapper.contains(message)).toEqual(true);
  });

  it("with open equals false, message and type props", () => {
    const wrapper = mount(
      <Snackbar open={false} message={message} type="success" />
    );

    expect(wrapper.props().open).toEqual(false);
    expect(wrapper.props().type).toEqual("success");
    expect(wrapper.props().message).toEqual(message);
    expect(wrapper.contains(message)).toEqual(false);
  });
});

describe("snapshots", () => {
  it("Snackbar snapshots", () => {
    const tree = shallow(<Snackbar open={true} message={message} />);

    expect(toJson(tree)).toMatchSnapshot();
  });
});
