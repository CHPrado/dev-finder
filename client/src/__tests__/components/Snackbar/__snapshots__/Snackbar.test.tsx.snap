// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots Snackbar snapshots 1`] = `
<WithStyles(ForwardRef(Snackbar))
  autoHideDuration={6000}
  onClose={[Function]}
  open={true}
>
  <WithStyles(ForwardRef(Alert))
    elevation={6}
    onClose={[Function]}
    severity="success"
    variant="filled"
  >
    Test message
  </WithStyles(ForwardRef(Alert))>
</WithStyles(ForwardRef(Snackbar))>
`;
